test/e2e/client_tls_test.go

```diff
@@ -384,6 +313,535 @@ func TestClientTLS(t *testing.T) {
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: crlConfigMap.Name,
+								},
+							},
+						},
+					}},
+				},
+			}
+
+			if err := kclient.Create(context.TODO(), &crlHostPod); err != nil {
+				t.Fatalf("Failed to create pod %q: %v", crlHostPod.Name, err)
+			}
+			defer assertDeletedWaitForCleanup(t, kclient, &crlHostPod)
+			crlHostService := corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "crl-host",
+					Namespace: namespace.Name,
+				},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{"app": crlHostName.Name},
+					Ports: []corev1.ServicePort{{
+						Name:       "http",
+						Port:       80,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromString("http-svc"),
+					}},
+				},
+			}
+			if err := kclient.Create(context.TODO(), &crlHostService); err != nil {
+				t.Fatalf("Failed to create service %q: %v", crlHostService.Name, err)
+			}
+			defer assertDeleted(t, kclient, &crlHostService)
+			// Wait for CRL host to be ready
+			err := wait.PollImmediate(2*time.Second, 3*time.Minute, func() (bool, error) {
+				if err := kclient.Get(context.TODO(), crlHostName, &crlHostPod); err != nil {
+					t.Logf("error getting pod %s/%s: %v", crlHostName.Namespace, crlHostName.Name, err)
+					return false, nil
+				}
+				for _, condition := range crlHostPod.Status.Conditions {
+					if condition.Type == corev1.PodReady && condition.Status == corev1.ConditionTrue {
+						return true, nil
+					}
+				}
+				return false, nil
+			})
+			// Create CA cert bundle
+			clientCAConfigmapName := "client-ca-cm-" + namespace.Name
+			clientCAConfigmap := corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clientCAConfigmapName,
+					Namespace: "openshift-config",
+				},
+				Data: map[string]string{
+					"ca-bundle.pem": strings.Join(tcCerts.CABundle, "\n"),
+				},
+			}
+			if err := kclient.Create(context.TODO(), &clientCAConfigmap); err != nil {
+				t.Fatalf("Failed to create CA cert configmap: %v", err)
+			}
+			defer assertDeleted(t, kclient, &clientCAConfigmap)
+			icName := types.NamespacedName{
+				Name:      "mtls-with-crls",
+				Namespace: operatorNamespace,
+			}
+			icDomain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+			ic := newPrivateController(icName, icDomain)
+			ic.Spec.ClientTLS = operatorv1.ClientTLS{
+				ClientCA: configv1.ConfigMapNameReference{
+					Name: clientCAConfigmapName,
+				},
+				ClientCertificatePolicy: operatorv1.ClientCertificatePolicyRequired,
+			}
+			if err := kclient.Create(context.TODO(), ic); err != nil {
+				t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+			}
+			defer assertIngressControllerDeleted(t, kclient, ic)
+
+			if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+				t.Fatalf("failed to observe expected conditions: %v", err)
+			}
+
```

Comment:
*nit* it's a nit pick, but probably worth a block comment here describing clientCerts are going to do, I tend to try to read comments first when deciphering code, feels more effcient:
```suggestion
// Create client certificate bundle as a configmap which we will mount into our test client pod
```